src/main/java/me/zeroeightsix/kami/module/modules/misc/HighwayTools.kt

```diff
@@ -0,0 +1,738 @@
+    var totalBlocksDestroyed = 0
+    private var totalBlocksDistance = 0
+
+    //val blockQueue = PriorityQueue<BlockTask>(TaskComparator())
+    var blockQueue: Queue<BlockTask> = LinkedList<BlockTask>()
+    private val doneQueue: Queue<BlockTask> = LinkedList<BlockTask>()
+    private var blockOffsets = mutableListOf<Pair<BlockPos, Boolean>>()
+    private var waitTicks = 0
+    private var blocksPlaced = 0
+    private lateinit var currentBlockPos: BlockPos
+    private lateinit var startingBlockPos: BlockPos
+
+    fun isDone(): Boolean { return blockQueue.size == 0 }
+
+    override fun onEnable() {
+        if (mc.player == null) {
+            disable()
+            return
+        }
+        buildDirectionSaved = getPlayerDirection()
+        startingBlockPos = BlockPos(floor(mc.player.posX).toInt(), floor(mc.player.posY).toInt(), floor(mc.player.posZ).toInt())
+        currentBlockPos = startingBlockPos
+        playerHotbarSlot = mc.player.inventory.currentItem
+        lastHotbarSlot = -1
+
+        playerHotbarSlot = mc.player.inventory.currentItem
+        buildDirectionCoordinate = if (buildDirectionSaved == 0 || buildDirectionSaved == 4) { startingBlockPos.getX() }
+        else if (buildDirectionSaved == 2 || buildDirectionSaved == 6) { startingBlockPos.getZ() }
+        else { 0 }
+
+        blockQueue.clear()
+        doneQueue.clear()
+        updateTasks()
+        printEnable()
+    }
+
+    override fun onDisable() {
+        if (mc.player == null) return
+
+        // load initial player hand
+        if (lastHotbarSlot != playerHotbarSlot && playerHotbarSlot != -1) {
+            mc.player.inventory.currentItem = playerHotbarSlot
+        }
+        if (isSneaking) {
+            mc.player.connection.sendPacket(CPacketEntityAction(mc.player, CPacketEntityAction.Action.STOP_SNEAKING))
+            isSneaking = false
+        }
+        playerHotbarSlot = -1
+        lastHotbarSlot = -1
+
+        BaritoneAPI.getProvider().primaryBaritone.pathingBehavior.cancelEverything()
+        printDisable()
+        totalBlocksPlaced = 0
+        totalBlocksDestroyed = 0
+        totalBlocksDistance = 0
+    }
+
+    override fun onUpdate() {
+        if (mc.playerController == null) return
+
+        if (!isDone()) {
+            if (!BaritoneAPI.getProvider().primaryBaritone.pathingBehavior.isPathing) {
+                pathing = false
+            }
+            if (!doTask()) {
+                if (!pathing && !ModuleManager.getModuleT(LagNotifier::class.java)!!.paused) {
+                    stuckDetector += 1
+                    blockQueue = LinkedList<BlockTask>(blockQueue.shuffled())
+                    if (stuckDetector > 20) {
+                        doneQueue.clear()
+                        blockQueue.clear()
+                        updateTasks()
+                    }
+                } else {
+                    doneQueue.clear()
+                    blockQueue.clear()
+                    updateTasks()
```

Comment:
I see you're clearing all this in multiple places, not just these, make a method